src/main/java/org/mule/extension/runtime/OperationContext.java

```diff
@@ -19,15 +19,32 @@
 {
 
     /**
-     * Returns the value associated to a parameter of name {@code parameterName}
+     * Returns the value associated to a parameter of name {@code parameterName}.
+     * <p/>
+     * {@code parameterName} might match a {@link ParameterModel} of the {@link OperationModel} being executed
+     * or can also be used to simply carry state across {@link Interceptor interceptors} and/or
+     * {@link OperationExecutor operation executors}
```

Comment:
I don't get this part: if a parameterName is not a parameter, why should this method return a value for it? Looks to me that we need a different way to pass state